src/nunavut/lang/cpp/support/serialization.j2

```diff
@@ -1,23 +1,67 @@
-/*
- *
- * UAVCAN serialization support routines.                                                                     +-+ +-+
- *  Copyright (C) 2014 Pavel Kirienko <pavel.kirienko@gmail.com>                                              | | | |
- *  Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.                                   \  -  /
- *  This software is distributed under the terms of the MIT License.                                            ---
- *                                                                                                               o
- * +------------------------------------------------------------------------------------------------------------------+
- */
-#ifndef NUNAVUT_SUPPORT_HPP_INCLUDED
-#define NUNAVUT_SUPPORT_HPP_INCLUDED
+{#-
+ # Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ # Copyright (C) 2021  UAVCAN Development Team  <uavcan.org>
+ # This software is distributed under the terms of the MIT License.
+ # Authors: David Lenfesty, Scott Dixon <dixonsco@amazon.com>, Pavel Kirienko <pavel@uavcan.org>,
+ #          Peter van der Perk <peter.vanderperk@nxp.com>
+-#}
+
+{%- macro assert(expression) -%}
+    {%- if options.enable_serialization_asserts -%}
+    NUNAVUT_ASSERT({{ expression }});
+    {%- endif -%}
+{%- endmacro -%}
+
+{%- macro float32_union() -%}
+    typedef union  // NOSONAR
+    {
+        uint32_t bits;
+        {{typename_float_32}} real;
+    } Float32Bits;
+{%- endmacro -%}
+
+// UAVCAN common serialization support routines.                                                             +-+ +-+
+// This file is based on canard_dsdl.h, which is part of Libcanard.                                          | | | |
```

Comment:
Is it though

Reply:
I'm not sure I understand your comment here?

Reply:
The file does not appear to have much in common with canard_dsdl.h, so maybe the comment is unnecessary. This is not very important though.